mathics/builtin/list/constructing.py

```diff
@@ -185,6 +185,14 @@ class Range(Builtin):
     def apply(self, imin, imax, di, evaluation):
         "Range[imin_?RealNumberQ, imax_?RealNumberQ, di_?RealNumberQ]"
 
+        if (
+            isinstance(imin, Integer)
+            and isinstance(imax, Integer)
+            and isinstance(di, Integer)
+        ):
+            result = [Integer(i) for i in range(imin.value, imax.value, di.value)]
+            return Expression(SymbolList, *result)
```

Comment:
In the expression refact branch one also has the option if indicating that *result is flat, fully evaluated, and ordered iff di > 0.